hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+         1. Reconstruct 1d or Nd hologram without reference
+         2. Reconstruct 1d or Nd hologram using single reference hologram
+         3. Reconstruct Nd hologram using Nd reference hologram (applies each reference to each hologram in Nd stack)
+
+         The reconstruction parameters (sb_position, sb_size, sb_smoothness) have to be 1d or to have same
+         dimensionality as the hologram.
+
+        Parameters
+        ----------
+        reference : ndarray, :class:`~hyperspy.signals.Signal2D, None
+            Vacuum reference hologram.
+        sb_size : float, :class:`~hyperspy.signals.BaseSignal, None
+            Sideband radius of the aperture in corresponding unit (see 'sb_unit'). If None,
+            the radius of the aperture is set to 1/3 of the distance between sideband and
+            centreband.
+        sb_smoothness : float, :class:`~hyperspy.signals.BaseSignal, None
+            Smoothness of the aperture in the same unit as sb_size.
+        sb_unit : str, None
+            Unit of the two sideband parameters 'sb_size' and 'sb_smoothness'.
+            Default: None - Sideband size given in pixels
+            'nm': Size and smoothness of the aperture are given in 1/nm.
+            'mrad': Size and smoothness of the aperture are given in mrad.
+        sb : str, None
+            Select which sideband is selected. 'upper' or 'lower'.
+        sb_position : tuple, :class:`~hyperspy.signals.Signal1D, None
+            Sideband position in pixel. If None, sideband is determined automatically from FFT.
+        output_shape: tuple, None
+            Choose a new output shape. Default is the shape of the input hologram. The output
+            shape should not be larger than the input shape.
+        plotting : boolean
+            Shows details of the reconstruction (i.e. SB selection).
+        show_progressbar : boolean
+            Shows progressbar while iterating over different slices of the signal (passes the parameter to map method).
+        log_parameters : boolean
+            Logs reconstruction parameters
+
+        Returns
+        -------
+        wave : :class:`~hyperspy.signals.WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+
+        Notes
+        -----
+
+        """
+
+        # Parsing reference:
+        if not isinstance(reference, HologramImage):
+            if isinstance(reference, Signal2D):
+                _logger.warning('The reference image signal type is not HologramImage. It will '
+                                'be converted to HologramImage automatically.')
+                reference.set_signal_type('hologram')
+            elif reference is not None:
+                reference = HologramImage(reference)
+
+        # Testing match of navigation axes of reference and self (exception: reference nav_dim=1):
+        if (reference and not reference.axes_manager.navigation_shape == self.axes_manager.navigation_shape
```

Comment:
The check is now made also before for the case if the input is HS signal. It is a little redundant in its current implementation but I think still less verbose than checking dimension of ndarray that can be an input as well.